src/KawaiiEngine/include/Engine.hpp

```diff
@@ -273,36 +282,21 @@ class Engine {
             static_cast<double>(std::chrono::duration_cast<std::chrono::microseconds>(dt_nano).count())
             / 1'000'000.0;
 
-        // note : camera logics should be a system // should be trigger by a signal
-        if (!ImGui::IsWindowFocused(ImGuiFocusedFlags_AnyWindow)) {
-            for (auto &camera : state->camera) {
-                const auto viewport = camera.getViewport();
-                const auto window_size = window->getSize<double>();
-
-                if (!Rect4<double>{
-                        static_cast<double>(viewport.x) * window_size.x,
-                        static_cast<double>(viewport.y) * window_size.y,
-                        static_cast<double>(viewport.w) * window_size.x,
-                        static_cast<double>(viewport.h) * window_size.y}
-                         .contains(state->mouse_pos)) {
-                    continue;
-                }
-
-                for (const auto &[button, pressed] : state->state_mouse_button) {
-                    if (!pressed) { continue; }
-                    camera.handleMouseInput(button, state->mouse_pos, state->mouse_pos_when_pressed, dt_secs);
-                }
-            }
-        }
-
-
         for (const auto &entity : world.view<Position3f, Velocity3f>()) {
             const auto &vel = world.get<Velocity3f>(entity);
             world.patch<Position3f>(entity, [&vel, &dt_secs](auto &pos) {
                 pos.component += vel.component * static_cast<double>(dt_secs);
             });
         }
 
+        // note : is it the best way of doing that ?
+        for (const auto &entity : world.view<Velocity3f, CameraData>()) {
+            const auto &vel = world.get<Velocity3f>(entity);
+            world.patch<CameraData>(entity, [&vel, &dt_secs](auto &cam) {
+                cam.target_center += vel.component * static_cast<double>(dt_secs);
+            });
+        }
```

Comment:
Target_center should be a child entity

Reply:
why ?

Reply:
If it is a separated entity (or a child even better), and have his own position, then the system velocity-position will be trigger automatically and we don t need this very specific system